src/modules/identity/RadixRemoteIdentity.ts

```diff
@@ -0,0 +1,178 @@
+import RadixECIES from '../crypto/RadixECIES'
+import RadixIdentity from './RadixIdentity'
+import RadixAccount from '../account/RadixAccount'
+import RadixSignature from '../atom/RadixSignature'
+import RadixSerializer from '../serializer/RadixSerializer'
+
+import { RadixAtom, RadixKeyPair } from '../RadixAtomModel'
+
+export default class RadixRemoteIdentity extends RadixIdentity {
+    private token: string
+    private remoteUrl: string
+
+    private signAtomSocket: WebSocket
+    private decryptECIESPayloadSocket: WebSocket
+
+    private constructor(readonly keyPair: RadixKeyPair, token: string, remoteUrl: string) {
+        super(keyPair)
+
+        this.token = token
+        this.remoteUrl = remoteUrl
+    }
+
+    /**
+     * Checks if a given web socket connection is still alive, otherwise creates a new one
+     * 
+     * @param _socket - WebSocket connection
+     * @returns A WebSocket connection
+     */
+    private getSocketConnection(_socket: WebSocket): WebSocket {
+        if (!_socket || _socket.readyState === _socket.CLOSING || _socket.readyState === _socket.CLOSED) {
+            _socket = new WebSocket(this.remoteUrl)
+        }
+        return _socket
+    }
+
+    /**
+     * Signs an atom with the wallet using the remote identity
+     * 
+     * @param atom - The atom to be signed
+     * @returns A promise with the signed atom
+     */
+    public signAtom(atom: RadixAtom): Promise<RadixAtom> {
+        return new Promise<RadixAtom>((resolve, reject) => {
+            const socket = this.getSocketConnection(this.signAtomSocket)
+
+            socket.onopen = () => {
+                socket.send(JSON.stringify({
+                    jsonrpc: '2.0',
+                    method: 'sign_atom',
+                    params: { 
+                        token: this.token,
+                        atom: atom.toJson(),
+                    },
+                    id: 0,
+                }))
+                socket.onmessage = (evt) => {
+                    atom.signatures = RadixSerializer.fromJson(JSON.parse(evt.data).result)
+                    resolve(atom)
+                }
+                socket.onerror = (error) => reject(`Error: ${JSON.stringify(error)}`)
```

Comment:
This is an odd way to reject with an error. I think the best practice is to always throw/reject with an Error object, see here for example https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject
What's the reason you've chosen to stringify it?

Reply:
If I don't stringify the error it's displayed as `[object Object]` instead of the expanded detailed view of the object.